cmd/metal-api/internal/datastore/poolsize.go

```diff
@@ -0,0 +1,105 @@
+package datastore
+
+import (
+	e "github.com/metal-stack/metal-api/cmd/metal-api/internal/eventbus"
+	"github.com/metal-stack/metal-api/cmd/metal-api/internal/metal"
+	"github.com/metal-stack/metal-lib/bus"
+	"go.uber.org/zap"
+)
+
+type manager struct {
+	rs          *RethinkStore
+	publisher   bus.Publisher
+	partitionid string
+	sizeid      string
+}
+
+func (m *manager) AllMachines() (metal.Machines, error) {
+	q := MachineSearchQuery{
+		PartitionID: &m.partitionid,
+		SizeID:      &m.sizeid,
+	}
+
+	allMachines := metal.Machines{}
+	err := m.rs.SearchMachines(&q, &allMachines)
+	if err != nil {
+		return nil, err
+	}
+
+	return allMachines, nil
+}
+
+func (m *manager) WaitingMachines() (metal.Machines, error) {
+	stateValue := string(metal.AvailableState)
+	notAllocated := true
+	q := MachineSearchQuery{
+		PartitionID:  &m.partitionid,
+		SizeID:       &m.sizeid,
+		StateValue:   &stateValue,
+		NotAllocated: &notAllocated,
+	}
+
+	waitingMachines := metal.Machines{}
+	err := m.rs.SearchMachines(&q, &waitingMachines)
+	if err != nil {
+		return nil, err
+	}
+
+	return waitingMachines, nil
+}
+
+func (m *manager) ShutdownMachines() (metal.Machines, error) {
+	sleeping := true
+	q := MachineSearchQuery{
+		PartitionID: &m.partitionid,
+		SizeID:      &m.sizeid,
+		Sleeping:    &sleeping,
+	}
+
+	shutdownMachines := metal.Machines{}
+	err := m.rs.SearchMachines(&q, &shutdownMachines)
+	if err != nil {
+		return nil, err
+	}
+
+	return shutdownMachines, nil
+}
+
+func (m *manager) Shutdown(machine *metal.Machine) error {
+	state := metal.MachineState{
+		Sleeping:    true,
+		Description: "shut down as exceeding maximum partition poolsize",
```

Comment:
The field was probably intended for a different purpose. Maybe we can introduce an individual struct like:

```golang
type Hibernation struct {
    Enabled bool
    Description string 
    Timestamp *time.Time
}
```